### PR TITLE
Enable image in table cells.

### PR DESCRIPTION
**Pull request by jodator**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Feature: Enable image in table cells.

---

### Additional information

* Part of: https://github.com/ckeditor/ckeditor5-table/pull/155

### CODE REVIEW

**Comment by J-Nagle** [comment]
Thanks very much for responding to our feedback in https://github.com/ckeditor/ckeditor5-table/pull/155#issuecomment-446797016. However, this gave us a `TypeError` when we went to test it. Attached is our build, below is the error, and here are links to the [configuration](https://pastebin.com/6WvB13au) and [stack trace](https://pastebin.com/xaJq9xLn). Hopefully this is enough...

```
TypeError: t.plugins.has is not a function
	at af.init (imageuploadediting.js:94)
InlineEditor.create.then.catch.error @ app.js:335 // When our system calls CKEditor
```
[CKEditor - Table image upload test build.zip](https://github.com/ckeditor/ckeditor5-image/files/2692701/CKEditor.-.Table.image.upload.test.build.zip)

**Comment by jodator** [comment]
@J-Nagle looks like latest changes in master :) I've updated PR (#t/ckeditor5-table/99) branches on
- ckeditor/ckeditor5-engine
- ckeditor/ckeditor5-widget

It should be OK but I'll test it better tomorrow morning as now I'm not able to properly test it.

Also I'll try to push work on this PR faster but nevertheless those changes will be publicly available in next release. Now it may fail as `master` branch is considered a working branch.

**Comment by J-Nagle** [comment]
Thanks for the fast response! The updated masters of Engine and Widget still don't make this work for us right now, but I absolutely appreciate the quick tweak nonetheless. Of course, we understand that other work may take priority, and will definitely upgrade to the next release when available. Cheers!

File: src/image/utils.js [hunk 1]
```diff
@@ -112,29 +112,28 @@ function isImageAllowedInParent( selection, schema, model ) {
...
+// Check if selection is on object.
 //
 // @returns {Boolean}
-function checkSelectionWithObject( selection, schema ) {
+function checkSelectionOnObject( selection, schema ) {
 	const selectedElement = selection.getSelectedElement();
 
-	const isSelectionOnObject = !!selectedElement && schema.isObject( selectedElement );
-	const isSelectionInObject = !![ ...selection.focus.getAncestors() ].find( ancestor => schema.isObject( ancestor ) );
+	return !( !!selectedElement && schema.isObject( selectedElement ) );
```
Comment (scofalik) [inline src/image/utils.js]: Doesn't this return the opposite thing? If this is checking if the selection is on object it should simply return:

`return selectedElement && schema.isObject( selectedElement );`

File: src/image/utils.js [hunk 2]
```diff
@@ -112,29 +112,28 @@ function isImageAllowedInParent( selection, schema, model ) {
...
+}
 
-	return !isSelectionOnObject && !isSelectionInObject;
+// Checks if selection is placed in other image (ie. in caption).
+function isInOtherImage( selection ) {
+	return [ ...selection.focus.getAncestors() ].every( ancestor => !ancestor.is( 'image' ) );
```
Comment (scofalik) [inline src/image/utils.js:128]: Simple optimization would be to check `.some` xD.

**Review by scofalik — CHANGES_REQUESTED** [review]
Small things.

File: src/image/utils.js (hunk 2 above)
Comment (jodator) [reply src/image/utils.js:128]: How would it be better then `.every`?

Both will return if condition is other then expected:
```js
.some( a => a.is( 'image' ) );
.every( a => !a.is( 'image' ) );
```

**Pull request merged**